### PR TITLE
FIX-#3767: Fix tests for Modin XGB Dmatrix

### LINKED ISSUE #3767: Fix dmatrix testing failures in CI
Starting from 4d7f6d4cfd2dc99d9bfd21bb62c5a74daa3ab256 `test_dmatrix` sporadically fails in CI testing. Possible reason is in strict check of equality between XGBoost/Modin XGBoost result:
https://github.com/modin-project/modin/blob/cc96f5d2c08b03f4bd5d3d261a6871b292d9230d/modin/experimental/xgboost/test/test_dmatrix.py#L129

### PR DESCRIPTION
**Pull request by prutskov**
Signed-off-by: Alexey Prutskov <alexey.prutskov@intel.com>

## What do these changes do?

This PR fixes periodic failures in CI on `experimental/xgboost/test/test_dmatrix.py`. The fix is in changing of random generated data for training/prediction purposes to existing known dataset.

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #3767  
- [x] tests added and passing
- [ ] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/experimental/xgboost/test/test_dmatrix.py
Comment (YarShev) [inline modin/experimental/xgboost/test/test_dmatrix.py:106]:
```suggestion
    preds = pandas.DataFrame(predictions).apply(lambda x: round(x))
```
I would prefer to be explicit so let's specify `axis` parameter for `apply`. Why do we need `round`?

File: modin/experimental/xgboost/test/test_dmatrix.py
Comment (YarShev) [inline modin/experimental/xgboost/test/test_dmatrix.py:107]: same comments as above

File: modin/experimental/xgboost/test/test_dmatrix.py
Comment (YarShev) [inline modin/experimental/xgboost/test/test_dmatrix.py:120]: Why do we need `repr` now?

File: modin/experimental/xgboost/test/test_dmatrix.py
Comment (prutskov) [reply modin/experimental/xgboost/test/test_dmatrix.py:96]: Because dataset with binary labels is used now

Comment (YarShev) [reply modin/experimental/xgboost/test/test_dmatrix.py:96]: Yes, I see. Why do we want to have a dataset for binary classification now?

Comment (prutskov) [reply modin/experimental/xgboost/test/test_dmatrix.py:96]: Previous data was incorrect. Binary classification is simply and fast.

File: modin/experimental/xgboost/test/test_dmatrix.py
Comment (prutskov) [reply modin/experimental/xgboost/test/test_dmatrix.py:120]: Modin XGB `predict` is async func. To get thrown exception here we should finish execution

File: modin/experimental/xgboost/test/test_dmatrix.py
Comment (vnlitvinov) [reply modin/experimental/xgboost/test/test_dmatrix.py:106]: also why don't you directly `.apply(round)` or `.apply(np.round)`?

File: modin/experimental/xgboost/test/test_dmatrix.py
Comment (YarShev) [reply modin/experimental/xgboost/test/test_dmatrix.py:96]: ok

File: modin/experimental/xgboost/test/test_dmatrix.py
Comment (YarShev) [reply modin/experimental/xgboost/test/test_dmatrix.py:120]: ok

File: modin/experimental/xgboost/test/test_dmatrix.py
Comment (prutskov) [reply modin/experimental/xgboost/test/test_dmatrix.py]: @YarShev, `axis` providing is added (I don't know if this makes sense, because we make `apply` for 1 column dataframe using non-reduce callable). `round` is need to determine class label (0 or 1) for each prediction value from column.
@vnlitvinov, I agree, `np.round` is used instead of lambda.

Comment (YarShev) [reply modin/experimental/xgboost/test/test_dmatrix.py]: I would also use pandas DataFrame for non Modin data here as I suggested earlier.

Comment (prutskov) [reply modin/experimental/xgboost/test/test_dmatrix.py]: No see the need in this, because in this test we don't compare pandas/modin dataframes functionality, we just use dataframe to apply `round`. But I don't mind, applied

**Comment by prutskov** [comment]
Blocked by #3772.

File: modin/experimental/xgboost/test/test_dmatrix.py
Comment (YarShev) [reply modin/experimental/xgboost/test/test_dmatrix.py]: If we had a bug in `apply`, we might have tests passed and wouldn't see an actual problem (even though we test `apply` separately).

**Comment by YarShev** [comment]
Merged #3773.

**Review by YarShev — APPROVED** [review]
LGTM assuming tests pass.

**Review by YarShev — APPROVED** [review]

**Pull request merged**
